educator/src/Dscp/Educator/Logic/Certificates.hs

```diff
@@ -84,16 +84,23 @@ ensureCertSubmissionExists = do
 -- | Add a certificate grade to database.
 addCertificateGrades
     :: (MonadIO m, MonadCatch m, Traversable t)
-    => CertificateMeta -> (t CertificateGrade) -> DBT 'WithinTx m (t PrivateTx)
-addCertificateGrades meta grades = do
+    => Hash Raw -> CertificateMeta -> (t CertificateGrade) -> DBT 'WithinTx m (t PrivateTx)
+addCertificateGrades rawHash meta grades = do
     ensureCertSubmissionExists
 
-    forM grades $ \grade -> do
-        submission <- liftIO randomCertSubmission
-        void $ createSignedSubmission submission
+    flip evalStateT True $ do
+        forM grades $ \grade -> do
+            injectHash <- get
+            put False
```

Comment:
Minor: I think written with `zip ... grades` that would be way shorter

Reply:
`grades` is `forall f. Foldable f => f Grade`, so I can't `zip` it. I'd want to.

Reply:
Right, harsh. :unamused: 
I think it's a fine reason to simplify the signature to `s/t/[]/` (if you see fit).